### ISSUE TITLE
project_by_component does not handle n-component tensors where n<3

### ISSUE THREAD

**Issue by scbarton** [sfepy/sfepy#389]
While solving a 2D problem I attempted to make a vector field variable from a two-component gradient vector field. The function `sfepy.descrete.projections.project_by_component` failed because it expected a 3-component input tensor:

```
    for ic in range(3):
        make_l2_projection_data(component, tensor_qp[..., ic, :].copy(),
                                order=order, ls=ls, nls_options=nls_options)
```

I was able to solve the problem for my own needs by checking for the dimension of the tensor:

```
    dim = nm.shape(tensor_qp)[2]
    
    for ic in range(dim):
        make_l2_projection_data(component, tensor_qp[..., ic, :].copy(),
                                order=order, ls=ls, nls_options=nls_options)
```

**Comment by rc** [comment]
Thanks! Would you like to submit a pull request? :)

**Comment by rc** [comment]
@scbarton, Could you try #391? It contains the fix you proposed.

**Comment by rc** [comment]
Added a test and merged #391, closing...
